### ISSUE TITLE
concat new data to $scope.data current top jumps down 7 rows after dataLoaded.

### ISSUE THREAD

**Issue by joel1618** [angular-ui/ui-grid#4047]
$scope.gridApi.infiniteScroll.saveScrollPercentage();
            $scope.gridOptions.data = $scope.gridOptions.data.concat(data);
            
            $scope.gridApi.infiniteScroll.dataLoaded();

After this code is used the grid loads the new data and keeps the position for the most part but actually jumps the top row down about 7 rows.  Is this a bug?  Or am I doing somethign wrong?  Thanks.

**Comment by PaulL1** [comment]
Your code looks like that in the infinite scroll tutorial: http://ui-grid.info/docs/#/tutorial/212_infinite_scroll.  And the infinite scroll tutorial doesn't exhibit that behaviour.  Are you able to create a plunker based on the infinite scroll tutorial and get it to exhibit the behaviour that you describe, or alternatively create a series of steps that I could use to recreate that behaviour on the infinite scroll tutorial?

**Comment by joel1618** [comment]
http://plnkr.co/edit/w04AmTxNZbDGA6xk9BGo?p=preview

modified the example to have more pages and to not remove old data (seems
to be some weirdness going on around that)

On the second "page" of the scroll you'll see it jump a bit and then a bit
more as you go to more pages.  If you notice on page 4,5,6+ the top row
will jump down about 7 rows or so and it stays at that jump size for the
rest of the infinite scroll.  The initial first scroll where it appears to
just append the data nd not move the scroll position is what would be
desired.

**Comment by kbolay** [comment]
I am seeing the same issue, I modified @joel1618 plunker to have a timeout so it is easier to see the jump of a few rows.

http://plnkr.co/edit/Osn12riNKoA4Y5tqhziP?p=preview

**Comment by eliasre** [comment]
Seeing the same issue as well.
The issue also seems to exist in the official infinite scroll example (table jumps when scrolling past 250).
http://ui-grid.info/docs/#/tutorial/212_infinite_scroll
I am currently working on an app that will be deployed to production and am unaware of good alternatives to ui-grid. Would be nice if this issue received attention.

**Comment by eliasre** [comment]
Hi @PaulL1.
Hope my comment wasn't considered as rude (that was not my intention).
Is there a chance that this issue will be looked at?
The issue seems to be general as it affects the tutorial example on the ui-grid page. Checked this on a couple of browsers on mac and windows.

**Comment by PaulL1** [comment]
No, the main issue is time.  I'm back full time working (more than full time) and we have a go live sequence from 1 Oct till 1 Nov.  And we're 3 weeks out from that.....so my time available for open source is basically nil.  I have seen some PRs go through on infinite scroll though, it's possible one or more of them fixes this.
